src/audio/volume/volume.c

```diff
@@ -276,7 +276,7 @@ static int volume_params(struct comp_dev *dev,
  * \param[in] chan Channel number.
  * \param[in] vol Target volume.
  */
-static inline int volume_set_chan(struct comp_dev *dev, int chan, uint32_t vol)
+static inline int volume_set_chan(struct comp_dev *dev, int chan, int32_t vol)
 {
```

Comment:
not clear why are you changing the type of vol? It doesn't seem to be related to this patch.

Reply:
A static analysis tool in CI nags about comparison of uint to less than zero if this is not done. There's no issues since VOL_MAX is much less than INT32_MAX.

I've not written this originally and IMHO it was a mistake to unsigned type for gain value. Negative inverting gain would be useful in some applications.

Reply:
@dbaluta This reminds me of our own code failing to compile with XCC with errors related to usage of unsigned types.

@singalsu I'm not sure how negative gain even works, can you explain? I feel like the actual effect over what you hear would be nothing audible... Just destructive interference in a further mixer component, maybe?

Reply:
Negative gain value just inverts. It could be useful with multi-way speakers and such. Some codecs/amplifiers may invert and some not. If gain can be negative then volume could align the polarity when needed. Though EQ can do it if need too.

Reply:
@paulstelian97 Ah yes, the mixer could subtract then too.

Reply:
> A static analysis tool in CI nags about comparison of uint to less than zero if this is not done. There's no issues since VOL_MAX is much less than INT32_MAX.
> 
> I've not written this originally and IMHO it was a mistake to use an unsigned type for the gain value. Negative inverting gain would be useful in some applications.

This kind of explanation I would like to be present in the commit message.

Reply:
I'll resolve this. Please let me know if not OK.